recipes/libcurl/all/conanfile.py

```diff
@@ -144,8 +148,6 @@ def configure(self):
 
         # These options are not used in CMake build yet
         if self._is_using_cmake_build:
-            del self.options.with_libidn
-            del self.options.with_librtmp
```

Comment:
Were those options just not handled by our Conan recipe or were they not available in the upstream CMake files? If the later is the case, you have to introduce a version switch based on which libcurl version added those options.

Reply:
dove into the git history, idn flag was added in 7.75, librtmp has been there since at least the earliest version the recipe supports (7.64.1).

I will add a version switch for libidn

Reply:
@Croydon can you please review again :)

Reply:
There is an issue with this `with_libidn` option, `requirements()` have not been updated to use `libidn` recipe.
Therefore it will link to system installed libidn, or just fail on Windows.

Reply:
For librtmp, it's worse, we don't even have a `librtmp` recipe.

Reply:
Sorry for not replying sooner, I was spending time with my family. Yeah, I left the behaviour of libidn/librtmp as it was and just exposed the options to CMake letting the libcurl CMakeLists figure that part out as it currently does.

A future PR to implement librtmp recipe and update this to pull from those recipes would definitely be an enhancement